include/ck_tile/ops/gemm/kernel/gemm_tile_partitioner.hpp

```diff
@@ -364,4 +364,466 @@ struct GemmSpatiallyLocalTilePartitioner
     index_t N;
 };
 
+enum StreamKReductionStrategy : uint32_t
+{
+    Atomic    = 0u,
+    Reduction = 1u
+};
+
+/**
+ * @brief Stream-K tile partitioner that dynamically balances work across workgroups
+ *
+ * This partitioner is responsible for mapping workgroups to tiles in the C tensor
+ * for the Stream-K algorithm which decomposes the GEMM problem
+ * into smaller work units and distributes them more evenly across available blocks,
+ * improving load balancing especially for cases where the K dimension is large.
+ */
+template <typename BlockGemmShape_,
+          StreamKReductionStrategy ReductionStrategy_ = StreamKReductionStrategy::Atomic,
+          uint32_t TileSwizzleSubM_                   = 8,
+          index_t GroupNum                            = 8,
+          index_t M01_                                = 4>
+struct StreamKTilePartitioner
+{
+    using BlockGemmShape = remove_cvref_t<BlockGemmShape_>;
+
+    static constexpr index_t MPerBlock          = BlockGemmShape::kM;
+    static constexpr index_t NPerBlock          = BlockGemmShape::kN;
+    static constexpr index_t KPerBlock          = BlockGemmShape::kK;
+    StreamKReductionStrategy reduction_strategy = ReductionStrategy_;
+    static constexpr index_t tile_swizzle_sub_m = TileSwizzleSubM_;
```

Comment:
```suggestion
template <typename BlockGemmShape,
          StreamKReductionStrategy ReductionStrategy = StreamKReductionStrategy::Atomic,
          uint32_t TileSwizzleSubM                   = 8,
          index_t GroupNum                            = 8,
          index_t M01                                = 4>
struct StreamKTilePartitioner
{
    static constexpr index_t MPerBlock          = BlockGemmShape::kM;
    static constexpr index_t NPerBlock          = BlockGemmShape::kN;
    static constexpr index_t KPerBlock          = BlockGemmShape::kK;
```

Meta-types should be non-const-ref already as they are passed in. I see this lots in CK code and I think we need to drop this practice. If in any case we find they are coming as const-ref let's find the source and correct it there before it's passed in. 

I also think it's a good idea to keep template parameters named in PascalCase so during the code we can see that variable has come from a static meta-data, such as a template argument.

Reply:
Also the template params GroupNum and M01 don't appear to be used anywhere in this class. Do we need them later?

Reply:
> Also the template params GroupNum and M01 don't appear to be used anywhere in this class. Do we need them later?

Those params were probably carried over from previous b2c maps in the original stream-k b2c map, but they're not used in stream-k